diplo.lua

```diff
@@ -0,0 +1,88 @@
+--[[
+diplo - Quick "Diplomacy"
+Without arguments:
+    Print list of civs the player civ has diplomatic relations to.
+    This list might not match the ingame civ list, but is what matters. Does not include 'no contact' civs, etc.
+    Also shows if relations match between player civ and other civ - a mismatch can potentially happen 'naturally', but is important when trying to make peace proper, as both civs need to be at peace with the other.
+With arguments:
+    diplo CIV_ID RELATION
+    Changes the relation to the civ identified by CIV_ID to the one specified in RELATION, making sure that this is mutual.
+    Relations: 0 = Peace; 1 = War; 3 = Alliance (seems to only apply to site govs...?)
+    Civ list is shown afterwards.
+]]
+
+local args = {...}
+
+-- player civ references:
+local p_civ_id = df.global.plotinfo.civ_id
+local p_civ = df.historical_entity.find(df.global.plotinfo.civ_id)
+
+-- get a very rough, but readable name for a civ:
+function get_raw_name(civ)
+    local raw_name = ""
+    for _, name_word in pairs(civ.name.words) do
+        if name_word ~= -1 then
+            raw_name = raw_name .. " " .. df.global.world.raws.language.words[name_word].word
+        end
+    end
+    raw_name = string.sub(raw_name, 2)
+    return raw_name
+end
+
+-- if no civ ID is entered, just output list of civs:
+if not args[1] then
+    goto outputlist

```

Comment:
though this works, I'd prefer if you made outputlist a function and called it (with a `return` statement afterwards to halt the script)

Reply:
Alright, I wasn't sure how viable gotos are in lua, but I can sure wrap that in a function.

Reply:
they just have the ability to create very confusing code. In DFHack, we try to keep the use of `goto` confined to the `continue` use case. e.g.:
```lua
for k,v in pairs(t) do
    if not condition(v) then goto continue end
    -- ... logic
    ::continue::
end
```

Reply:
Refactored script to use functions, `goto` removed.